tests/integration_tests/test_sklearn.py

```diff
@@ -302,6 +312,34 @@ def test_objective_error_score_invalid() -> None:
         optuna_search.fit(X)
 
 
+@pytest.mark.parametrize(
+    "param_dist, expect",
+    [
+        ({"max_depth": distributions.IntDistribution(1, 10)}, False),
+        ({"max_depth": distributions.IntDistribution(0, 0)}, True),
+    ],
+)
+@pytest.mark.filterwarnings("ignore::RuntimeWarning")
+@pytest.mark.filterwarnings("ignore::UserWarning")
+def test_objective_error_score_nan_regression(param_dist: dict, expect: bool) -> None:
+    X, y = make_regression(n_samples=100, n_features=10)
+    est = DecisionTreeRegressor()
+    optuna_search = integration.OptunaSearchCV(
+        est,
+        param_dist,
+        cv=3,
+        error_score=np.nan,
+        random_state=0,
+    )
```

Comment:
I guess what you want to do is as follows:

```suggestion
        ({"max_depth": distributions.IntDistribution(0, 1)}, False),
        ({"max_depth": distributions.IntDistribution(0, 0)}, True),
    ],
)
@pytest.mark.filterwarnings("ignore::RuntimeWarning")
@pytest.mark.filterwarnings("ignore::UserWarning")
def test_objective_error_score_nan_regression(param_dist: dict, expect: bool) -> None:
    X, y = make_regression(n_samples=100, n_features=10)
    est = DecisionTreeRegressor()
    study = optuna.create_study()
    optuna_search = integration.OptunaSearchCV(
        est,
        param_dist,
        cv=3,
        error_score=np.nan,
        random_state=0,
        study=create_study(sampler=BruteForceSampler(), direction="maximize"),
    )
```